apps/web/src/common/modules/navigations/gnb/modules/gnb-profile/GNBProfile.vue

```diff
@@ -1,6 +1,162 @@
+
+import { vOnClickOutside } from '@vueuse/components';
+import {
+    computed, getCurrentInstance, reactive,
+} from 'vue';
+import type { Location } from 'vue-router';
+import type { Vue } from 'vue/types/vue';
+
+import {
+    PI, PDivider, PButton,
+} from '@spaceone/design-system';
+import type { MenuItem } from '@spaceone/design-system/types/inputs/context-menu/type';
+import ejs from 'ejs';
+
+import { store } from '@/store';
+import { i18n } from '@/translations';
+
+import { CURRENCY_SYMBOL } from '@/store/modules/settings/config';
+import { languages } from '@/store/modules/user/config';
+
+import config from '@/lib/config';
+import { showSuccessMessage } from '@/lib/helper/notice-alert-helper';
+
+import ErrorHandler from '@/common/composables/error/errorHandler';
+
+import { AUTH_ROUTE } from '@/services/auth/route-config';
+import { INFO_ROUTE } from '@/services/info/route-config';
+import { MY_PAGE_ROUTE } from '@/services/my-page/route-config';
+
+interface Props {
+    visible: boolean
+}
+
+const props = withDefaults(defineProps<Props>(), {
+    visible: false,
+});
+
+const emit = defineEmits<{(e: 'update:visible', visible: boolean): void; }>();
+const vm = getCurrentInstance()?.proxy as Vue;
+
+const state = reactive({
+    userIcon: computed(() => {
+        if (state.isDomainOwner) return 'img_avatar_root-account';
+        if (state.hasDomainRole) return 'img_avatar_admin';
+        return 'img_avatar_user';
+    }),
+    name: computed(() => store.state.user.name),
+    email: computed(() => store.state.user.email),
+    role: computed(() => {
+        const roleArray = store.getters['user/roleNames'];
+        return roleArray.join(', ');
+    }),
+    language: computed(() => store.getters['user/languageLabel']),
+    currency: computed(() => store.state.settings.currency),
+    timezone: computed(() => store.state.user.timezone),
+    domainId: computed(() => store.state.domain.domainId),
+    userId: computed(() => store.state.user.userId),
+    hasDomainRole: computed((() => store.getters['user/hasDomainRole'])),
+    isDomainOwner: computed(() => store.getters['user/isDomainOwner']),
+    hasPermission: computed(() => store.getters['user/hasPermission']),
+    languageMenuVisible: false,
+    currencyMenuVisible: false,
+    supportedMenu: computed(() => {
+        const docsList = config.get('DOCS') ?? [];
+        const data = { lang: store.state.user.language };
+        return docsList.map((d) => ({
+            label: ejs.render(d?.label ?? '', data),
+            link: ejs.render(d?.link ?? '', data),
+        }));
+    }),
+    languageMenu: computed(() => Object.entries(languages).map(([k, v]) => ({
+        label: v, name: k,
+    }))),
+    currencyMenuItems: computed<MenuItem[]>(() => Object.keys(store.state.settings.currencyRates).map((currency) => ({
+        name: currency,
+        label: `${CURRENCY_SYMBOL[currency]}${currency}`,
+    }))),
+    profileMenuRef: null as null|HTMLElement,
```

Comment:
nitpick: ref in state should be changed to `Ref` object. So I recommend using `toRef` or `Ref`~!